sale_product_approval/models/product_product.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    sale_ok = fields.Boolean(
+        compute="_compute_sale_ok_product",
+        string="Can be Sold",
+        default=False,
+        store=True,
+    )
+    candidate_sale = fields.Boolean(
+        string="Candidate to be Sold",
+        default=True,
+    )
+
+    @api.depends("product_state_id")
+    def _compute_sale_ok_product(self):
+        for product in self:
+            if product.product_state_id:
+                product.sale_ok = (
+                    product.candidate_sale and product.product_state_id.approved_sale
+                )
+                if not product.sale_ok:
+                    product_variant_ids = (
+                        self.env["product.product"]
+                        .search([("product_tmpl_id", "=", product.id)])
+                        .ids
+                    )
+                    order_ids = (
+                        self.env["sale.order.line"]
+                        .search(
+                            [
+                                ("product_id", "in", product_variant_ids),
```

Comment:
`product_variant_ids` is available as a One2many field
```suggestion
                                ("product_id", "in", product.product_variant_ids.ids),
```